test/functional/feature_assumeutxo.py

```diff
@@ -247,6 +254,15 @@ def check_tx_counts(final: bool) -> None:
 
         assert_equal(n1.getblockchaininfo()["blocks"], SNAPSHOT_BASE_HEIGHT)
 
+        self.log.info("Submit a stale block that forked off the chain before the snapshot")
```

Comment:
85233b203231f39c4436017860248ff9ad08ad42: This is the test mentioned in the PR description? Maybe introduce it in a separate commit so it's a bit easier to test the bug fix.

When I undo the changes in `chain.h` and `validation.cpp` the test fails earlier (line 243), so it tricky to test this part now.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29370#discussion_r1520020544

> [85233b2](https://github.com/bitcoin/bitcoin/commit/85233b203231f39c4436017860248ff9ad08ad42): This is the test mentioned in the PR description? Maybe introduce it in a separate commit so it's a bit easier to test the bug fix.

Good idea, I moved the tests for the two crashes into separate commits.

Reply:
Nice, I was able to produce the crashes - by reverting the main commit and then keeping _one_ of the two crash test commits. Didn't look at the crash log itself.